### PR TITLE
Feat/new init api

### PR DESCRIPTION
**Pull request by oed**
This PR adds new apis as specified in https://github.com/3box/3box-js/issues/640

To be merged after #667

### CODE REVIEW

File: src/3box.js [hunk 1]
```diff
@@ -167,6 +208,28 @@ class Box extends BoxApi {
...
+   * @param     {String}    opts.firstModerator     DID of first moderator of a thread, by default, user is first moderator
+   * @param     {Boolean}   opts.members            join a members only thread, which only members can post in, defaults to open thread
+   * @param     {Boolean}   opts.noAutoSub          Disable auto subscription to the thread when posting to it (default false)
+   * @param     {Boolean}   opts.ghost              Enable ephemeral messaging via Ghost Thread
+   * @param     {Number}    opts.ghostBacklogLimit  The number of posts to maintain in the ghost backlog
+   * @param     {Array<Function>} opts.ghostFilters Array of functions for filtering messages
+   *
+   * @return    {Thread}                  An instance of the thread class for the joined thread
+   */
+  async joinThread (space, name, opts) {
```
Comment (zachferland) [inline src/3box.js]: seems confusing to have this func here with the same name, considered have openThread and joinThread or still in a space like space.openThread? but i guess if in space then space with need an open/auth flow, so maybe here but  openThread

File: src/3box.js [hunk 2]
```diff
@@ -221,6 +284,7 @@ class Box extends BoxApi {
    * @property {String} DID        the DID of the user
    */
   get DID () {
+    if (!this._3id) throw new Error('DID: auth required')
```
Comment (zachferland) [inline src/3box.js]: could probs add this constraint to some func/class modifier/wrap, but probably clearer and simpler to keep as is for now

Comment (oed) [reply src/3box.js]: yeah, effort vs gain :)

File: src/3box.js (hunk 1 above)
Comment (oed) [reply src/3box.js]: So just rename this fn to `openThread`? The thing that might become confusing is to have two different names for two methods that essentially does the same. Note that if you join a thread from here and then call `box.auth` for the given space you will automatically be able to post in this thread.

Comment (oed) [reply src/3box.js]: On second thought, `openThread` here makes alot of sense. Going to change it!

**Comment by oed** [comment]
Thanks @zachferland I'd say wait to run tests until we prepare the release branch :)

**Pull request merged**